jbi/models.py

```diff
@@ -304,11 +306,15 @@ def extract_from_see_also(self):
 
     def lookup_action(self, actions: Actions) -> Action:
         """Find first matching action from bug's whiteboard list"""
-        tags: list[str] = self.get_potential_whiteboard_config_list()
-        for tag in tags:
-            if action := actions.get(tag):
-                return action
-        raise ActionNotFoundError(", ".join(tags))
+        if self.whiteboard:
+            for tag, action in actions.by_tag.items():
+                pre = r"\[.*" if action.brackets_required else ""
+                post = r".*\]" if action.brackets_required else ""
+                search_string = f"{pre}{tag.lower()}{post}"
```

Comment:
`[.*{tag}.*]` may be too permissive:

```
>>> import re
>>> re.search(r"\[.*devtest.*\]", "[foo] devtest [bar]")
<re.Match object; span=(0, 19), match='[foo] devtest [bar]'>
>>>
```

you would have to change it to something like this:

```
>>> re.search(r"\[[^\]]*devtest[^\]]*\]", "[foo] devtest [bar]")
>>>
>>> re.search(r"\[[^\]]*devtest[^\]]*\]", "[foo-devtest-bar]")
<re.Match object; span=(0, 17), match='[foo-devtest-bar]'>
>>>
```
≈